### ISSUE TITLE
Portable installation option (Windows)

### ISSUE THREAD

**Issue by ghost** [modrinth/code#501]
### Is your suggested enhancement related to a problem? Please describe.

I don't like using installers, because zips give me more control in where and how I install software.

### Describe the solution you'd like

Add a .zip portable download option for Windows

**Comment by TheMCSpacePro** [comment]
I second this, since not everyone has the permissions to run the installer.

**Comment by Bobrokus** [comment]
> I second this, since not everyone has the permissions to run the installer.

Plus it's easier to maintain, move data, purge data and would allow easy installation via [Scoop](scoop.sh). Portability is the GOAT

**Comment by brawaru** [comment]
I mean it's *kind of* possible https://github.com/brawaru/bucket/blob/main/bucket/modrinth-app.json :)

**Comment by Bobrokus** [comment]
> I mean it's _kind of_ possible [brawaru/bucket@`main`/bucket/modrinth-app.json](https://github.com/brawaru/bucket/blob/main/bucket/modrinth-app.json?rgh-link-date=2024-07-09T14%3A20%3A05Z) :)

I'm currently working on my own manifest and I see we chose a similar approach, though many improvements could be made in your manifest. For example, you don't need to set the env variable in a bat file. You can set it as a user env variable via `env_set` in the manifest.

So yes it is possible but I would appreciate a fully self-contained version without having to create a new env variable.

**Comment by brawaru** [comment]
> You can set it as a user env variable via env_set in the manifest.

Yeeeah but I don't want that, because that would mean that any Modrinth App running on my computer will use that directory, so I just set it only for the one installed by Scoop.

**Comment by Bobrokus** [comment]
> > You can set it as a user env variable via env_set in the manifest.
> 
> Yeeeah but I don't want that, because that would mean that any Modrinth App running on my computer will use that directory, so I just set it only for the one installed by Scoop.

Hey! I understand your concern and I've replied to you: https://github.com/modrinth/code/issues/497#issuecomment-2218453703

**Comment by Bobrokus** [comment]
I suggest renaming this issue to "Self-contained install" as it conveys the idea better imo

**Comment by FireController1847** [comment]
+1 on this. It would be very useful to be able to run this app from a Flash Drive or Portable SSD.

**Comment by regorxxx** [comment]
Also interested on this

**Comment by MiguVT** [comment]
Hi! I really think a fully portable installation option would be amazing for the Modrinth app, especially for easy portability on USB drives. I’ll wait to see if this gets accepted into the original Modrinth project, but if not, I plan to work on adding it myself in my fork. Before that, I’ll submit a PR to check if the only reason this hasn’t been done yet is that no contributor has picked it up.

**Comment by MiguVT** [comment]
Done! I decided to make it by myself soo no more wait! (I didnt know this was that old) I will wait it to get accepted.

https://github.com/modrinth/code/pull/4171

**Comment by Mariosonicman91** [comment]
> Done! I decided to make it by myself soo no more wait! (I didnt know this was that old) I will wait it to get accepted.
> 
> [#4171](https://github.com/modrinth/code/pull/4171)

GAHHH THANK YOU

**Comment by ghost** [comment]
> Done! I decided to make it by myself soo no more wait! (I didnt know this was that old) I will wait it to get accepted.
> 
> [#4171](https://github.com/modrinth/code/pull/4171)

All that work and what did it get you :(

**Comment by regorxxx** [comment]
I'm also failing to understand why it was closed without a review or change requests. 

I mean, if the reason is "the devs would do other thing", then the idea is to see such thing in practice... or ask for changes. 

This issue has been opened for 2 years btw, and the only contribution related to it is closed without alternative?

**Comment by ghost** [comment]
Only way to have an actual portable modrinth is using MiguTV's Migurinth and creating an empty portable.txt file in it and it will become a portable version of Modrinth

**Comment by MiguVT** [comment]
I understand the reason why Modrinth closed it, it's not because "the devs would do other thing". They could just give the reason (since this is their app and they have the most control over it). The real problem was that my workaround isn't a real solution.

What I'm trying to call a "real solution" is a clean way to reach that. The issue is that Modrinth is based on [Tauri](https://v2.tauri.app/), and Tauri doesn't have proper support for portable apps.

The only way to make a portable mode in Modrinth without a workaround would be to look for a [Tauri](https://v2.tauri.app/) alternative, which could be very difficult and require a lot of work.

Basically, they want to follow KISS (Keep it simple, stupid) practices.
